src/ngraph/op/fused/lstm_cell.cpp

```diff
@@ -0,0 +1,352 @@
+{
+    // ------ VARIABLE'S NAMES AND ACRONYM DEFINITIONS ------
+    // The names used below are analogous to the one used in ONNX documentation.
+    //
+    // ------ ACRONYMS ------
+    // i - input gate
+    // o - output gate
+    // f - forget gate
+    // c - cell gate
+    // t - time step (t-1 means previous time step)
+    // W  - W parameter weight matrix for input, output, forget, and
+    //      cell gates.
+    // R  - R recurrence weight matrix for input, output, forget, and
+    //      cell gates.
+    // Wb - W bias vectors for input, output, forget, and cell gates.
+    // Rb - R bias vectors for input, output, forget, and cell gates.
+    // ------ VARIABLE NAMES ------
+    // p_[iof] - P peephole weight vector for respectively: input, output,
+    //           and forget gates.
+    // Xt_W    - Input sequence multiplied by weights tensor at current time
+    //           step.
+    // Ht_R    - Hidden state multiplied by weights tensor at current time step.
+
+    // (.) - Denotes element-wise multiplication.
+    // *   - Denotes dot product.
+
+    // ---- Equations ----
+    // f, g, h - are activation functions.
+    // it = f(Xt*(Wi^T) + Ht-1*(Ri^T) + Pi (.) Ct-1 + Wbi + Rbi)
+    // ft = f(Xt*(Wf^T) + Ht-1*(Rf^T) + Pf (.) Ct-1 + Wbf + Rbf)
+    // ct = g(Xt*(Wc^T) + Ht-1*(Rc^T) + Wbc + Rbc)
+    // Ct = ft (.) Ct-1 + it (.) ct
+    // ot = f(Xt*(Wo^T) + Ht-1*(Ro^T) + Po (.) Ct + Wbo + Rbo)
+    // Ht = ot (.) h(Ct)
+    // --------------------
+
+    const auto& p_i = m_p_iof.at(0);
+    const auto& p_o = m_p_iof.at(1);
+    const auto& p_f = m_p_iof.at(2);
+
+    // Xt*(W^T) -- for [iofc] gates.
+    auto Xt_W = std::make_shared<ngraph::op::Dot>(m_X, ngraph::op::util::transpose(m_W));
+    // Ht-1*(R^T)  -- for [iofc] gates.
+    auto Ht_R = std::make_shared<ngraph::op::Dot>(m_H_t, ngraph::op::util::transpose(m_R));
+    // Xt*(W^T) + Ht-1*(R^T) + Wb + Rb  -- for [iofc] gates.
+    auto gates = add(Xt_W, add(Ht_R, m_bias));
+
+    NodeVector split_gates = builder::split(gates, 4, -1);
+    auto i_t = split_gates.at(0);
+    auto o_t = split_gates.at(1);
+    auto f_t = split_gates.at(2);
+    auto c_t = split_gates.at(3);
+
+    // f(Xt*(Wi^T) + Ht-1*(Ri^T) + Pi (.) Ct-1 + Wbi + Rbi)
+    i_t = m_activation_f(clip(add(i_t, mul(p_i, m_C_t)), get_clip()));
+    if (m_input_forget)
+    {
+        // Couple input with forget gate: 1 - i_t
+        f_t =
+            sub(ngraph::op::Constant::create(i_t->get_element_type(),
+                                             i_t->get_shape(),
+                                             std::vector<float>(shape_size(i_t->get_shape()), 1.f)),
+                i_t);
+    }
+    else
+    {
+        // f(Xt*(Wf^T) + Ht-1*(Rf^T) + Pf (.) Ct-1 + Wbf + Rbf)
+        f_t = m_activation_f(clip(add(f_t, mul(p_f, m_C_t)), get_clip()));
+    }
+    // ft (.) Ct-1 + it (.) ct
+    auto C = add(mul(f_t, m_C_t), mul(i_t, m_activation_g(clip(c_t, get_clip()))));
```

Comment:
Please consider fbroadcast data to the biggest one first and then use standard operators +-* for all of these equations.

Reply:
I think I will wait until #2936 will be merged. When broadcasting will happen automatically I'll change it to operators.